### PR TITLE
Cleaned up some ForkChoice functions, added a check to make sure FC is initialised

### PR DESCRIPTION
**Pull request by rolfyone**
## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 1]
```diff
@@ -179,30 +181,128 @@ public void onForkChoiceUpdatedResult(
...
+    performanceRecord.ifPresent(TickProcessingPerformance::tickProcessorComplete);
+    final UInt64 currentSlot = spec.getCurrentSlot(store);
+    if (currentSlot.isGreaterThan(slotAtStartOfTick)) {
+      applyDeferredAttestations(currentSlot).ifExceptionGetsHereRaiseABug();
+    }
+    performanceRecord.ifPresent(TickProcessingPerformance::deferredAttestationsApplied);
+  }
+
+  private void initializeProtoArrayForkChoice() {
+    initialized = true;
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java]: I'd put `volatile` here, to be super safe

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 2]
```diff
@@ -312,47 +312,63 @@ private SafeFuture<Boolean> processHead(
...
+                          justifiedCheckpoint);
                       notifyForkChoiceUpdatedAndOptimisticSyncingChanged(
                           isPreProposal ? nodeSlot : Optional.empty());
-
                       return true;
                     }));
   }
 
+  private void updateHeadTransaction(
+      Optional<UInt64> nodeSlot,
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java]: `final`s :)

**Review by tbenr — APPROVED** [review]
LGTM. just a couple of comments

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java]: removed the idea anyway...

**Pull request merged**